File: kotlin/pangram/src/main/kotlin/Pangram.kt
package Pangram

import java.util.HashSet

const val ALPHABET_SIZE = 26

fun isPangram(s: String): Boolean {
    val seen = HashSet<Char>()

    for (c: Char in s.toLowerCase()) {
        if (c >= 'a' && c <= 'z') seen.add(c)
    }

    return seen.size == ALPHABET_SIZE
}

File: cpp/pangram/pangram.cpp
#include "pangram.h"
#include <bitset>
#include <locale>

namespace pangram {

constexpr auto letters_count = 'z' - 'a' + 1;

char normalize(const char x) { return std::tolower(x, std::locale::classic()); }

bool is_letter(const char x) {
    const char nx = normalize(x);
    return nx >= 'a' && nx <= 'z';
}

bool is_pangram(const std::string &xs) {
    std::bitset<letters_count> seen;

    for (const auto x : xs) {
        if (is_letter(x)) {
            seen[normalize(x)] = true;

            if (seen.count() == letters_count)
                return true;
        }
    }

    return false;
}

} // namespace pangram
